c7n/commands.py

```diff
@@ -232,12 +236,32 @@ def validate(options):
                 )
             ))
         used_policy_names = used_policy_names.union(conf_policy_names)
+        source_locator = None
+        if fmt[0] == "y":
+            # For yaml files there is at least the expectation that the policy
+            # name is on a line by itself. With JSON, the file could be one big
+            # line. At this stage we are only attempting to find line number for
+            # policies in yaml files.
+            source_locator = SourceLocator(config_file)
```

Comment:
for reference directionally we want the policy object to know its source, but not sure we're ready for that. the loader class does reference a uri to policies but its not yet universal in usage.

Reply:
I've added a PR #6167 that would add file and line metadata to policies. This would mean we'd not need the SourceLocator at all, and we could just add the filename and line number details if it exists in the policy metadata.

Reply:
we still want a source locator, we don't want to do upfront work in loader without cause, and source lines are only needed rarely wrt to loader usage.